cirq/experiments/fidelity_estimation.py

```diff
@@ -0,0 +1,51 @@
+"""Estimate fidelity of large random quantum circuit from observed bitstrings.
+
+Fidelity estimator defined here is used in cross-entropy benchmarking and
+works under the assumption that the evaluated circuit is sufficiently
+random, see https://arxiv.org/abs/1608.00263.
+"""
+
+from typing import cast, List, Sequence, Tuple
+
+import numpy as np
+
+from cirq.circuits import Circuit
+from cirq.ops import Qid
+from cirq.sim import Simulator, WaveFunctionTrialResult
+
+
+def estimate_circuit_fidelity(circuit: Circuit, qubit_order: Sequence[Qid],
+                              bitstrings: Sequence[int]) -> float:
```

Comment:
I think the bitstrings should be input in the same format as the output of running a circuit on a sampler (simulator or Engine), that is, as a numpy array. That way we don't need to perform any conversions. Also, I don't think `qubit_order` is necessary because the measurement gates in the circuit already determine a qubit order so we can assume that the bitstrings use the same order.

Reply:
Re numpy array: Done.

Re qubit_order: I don't think I can use `cirq.final_wavefunction()` with measurement gates - it'll collapse the wave function.

Reply:
I meant a 2-d numpy array of bools, to match with the output of `Sampler` and how bitstrings are stored in `TrialResult.measurements`.

Reply:
Personally I strongly prefer the bit packing into integers, as is done in `TrialResults.data`. We should continue moving away from the double numpy array format, which I think has been confusing to users.

Reply:
Ok in that case you should change the input type back to Sequence[int].

Reply:
Yeah, I also prefer 1-D integer array. They render in a much more readable way on screen and make direct access and iteration simpler for most use-cases.

I'll keep int array format in this PR and will make changes elsewhere to be consistent with this representation.

Reply:
FTR: #2151.